packages/p2p-v2/src/pages/orders/components/OrderDetailsCard/OrderDetailsCardReview/OrderDetailsCardReview.tsx

```diff
@@ -0,0 +1,86 @@
+        is_reviewable: isReviewable,
+        isCompletedOrder,
+        review_details: reviewDetails,
+    } = orderDetails;
+    const { data: p2pSettingsData } = p2p.settings.useGetSettings();
+    const [remainingReviewTime, setRemainingReviewTime] = useState<string | null>(null);
+    const ratingAverageDecimals = reviewDetails ? Number(Number(reviewDetails.rating).toFixed(1)) : 0;
+    const { isMobile } = useDevice();
+    const textSize = isMobile ? 'sm' : 'xs';
+
+    useEffect(() => {
+        if (completionTime && p2pSettingsData?.review_period) {
+            setRemainingReviewTime(getDateAfterHours(completionTime, p2pSettingsData.review_period));
+        }
+    }, [completionTime, p2pSettingsData?.review_period]);
+
+    if (isCompletedOrder && !hasReviewDetails)
+        return (
+            <div className='flex flex-col px-[1.6rem] py-10 gap-3'>
+                <Button
+                    className='border-[1px] gap-[0.2rem] pl-4 pr-[1.4rem] w-fit'
+                    color='black'
+                    disabled={!isReviewable}
+                    icon={<StandaloneStarFillIcon fill='#FFAD3A' height={18} width={18} />}
+                    variant='outlined'
+                >
+                    <Text size={textSize}>{isReviewable ? 'Rate this transaction' : 'Not rated'}</Text>
+                </Button>
+                <Text color='less-prominent' size={isMobile ? 'xs' : '2xs'}>
+                    {isReviewable
+                        ? `You have until ${remainingReviewTime} GMT to rate this transaction.`
+                        : 'You can no longer rate this transaction.'}
+                </Text>
+            </div>
+        );
+
+    if (hasReviewDetails) {
+        return (
+            <div className='flex flex-col px-[1.6rem] py-10 gap-4'>
+                <Text weight='bold'>Your transaction experience</Text>
+                <div className='flex justify-between w-4/5 ml-2'>
+                    <StarRating isReadonly ratingValue={ratingAverageDecimals} starsScale={1.2} />
+                    <Text as='div' className='flex items-center gap-1' color='less-prominent' size={textSize}>
+                        {reviewDetails?.recommended !== null &&
+                            (reviewDetails?.recommended ? (
+                                <>
+                                    <StandaloneThumbsUpRegularIcon
+                                        className='mb-[0.3rem]'
+                                        fill='#4BB4B3'
+                                        iconSize='sm'
+                                    />
+                                    Recommended
+                                </>
+                            ) : (
+                                <>
+                                    <StandaloneThumbsDownRegularIcon fill='#ec3f3f' iconSize='sm' />
+                                    Not Recommended
+                                </>
+                            ))}
```

Comment:
@ameerul-deriv, perhaps it's better to extract this out to it's own component, because It looks a lot like an independent component (something like `<RecommendationStatus>`), and also, having these will definitely increase the cognitive complexity later on as this component grows. 🤔 
```suggestion
                        {reviewDetails?.recommended !== null &&
                            (reviewDetails?.recommended ? (
                                <>
                                    <StandaloneThumbsUpRegularIcon
                                        className='mb-[0.3rem]'
                                        fill='#4BB4B3'
                                        iconSize='sm'
                                    />
                                    Recommended
                                </>
                            ) : (
                                <>
                                    <StandaloneThumbsDownRegularIcon fill='#ec3f3f' iconSize='sm' />
                                    Not Recommended
                                </>
                            ))}
```

Reply:
Hmmm we could extract this but there aren't any other components or pages within P2P where this is shown besides here in order details 😅 I think its better to leave this here since its only used here. And as for if this component will grow and become bigger, right now there are no other future features lined up that will affect this component so the complexity won't be an issue

Reply:
Alright @ameerul-deriv, but It's always best to separate concerns whenever you can. Consider this as defensive programming, but from UI perspective. 
Having this would like I mentioned before, reduces cognitive complexity (which will eventually increase), It also makes the overall component easier to understand and maintain. But like they said, it's up to you guys as a squad. So feel free to resolve this conversation and let your squad members approve the PR 👍

Reply:
Agree. it might not be reusable, but its nice to have smaller components

Reply:
hmmm in this case we got to do a lot in P2PV2 tbh, theres a lot of giant components such as the Forms, Modals, Etc. It really depends on the scope of the Component IMO. Relatively this component is quite small in comparison to the much bigger components in our app. I understand your concerns with seperation of concerns but the conditional ternary is only one level, and not nested, and the if check for recommended is not null is due to the possibilities of the user not recommending the order details. 

In future, we could perhaps follow the methadologies of Atomic Design Patterns as this would resolve a lot of the current structural and complexity of some components within our app. We can discuss this further, in the channel, but for now I don't think we need to really create another component for this

Reply:
Alright @ameerul-deriv, but do all those places have nested conditionals? 🤔 I specifically picked this place out because it's a hotspot for a lot of errors. You would need to document with TSDoc how the logic is in that conditional, so that the people who come before you can easily understand what's going on in the component without too much effort. Bare in mind that this doesn't really need to be in a separate file, but can be embedded in this file as a helper component. but like I told you, no need to drag this further a long. Just resolve the conversation and let other people approve.

Reply:
Hmm its not really a nested conditional, it's single-level conditional 🤔 it would need to have another conditional isnide of the ternary operator for it to be nested right? Also I don't think this logic isn't particularly complex as its just checking for nullish values to render the ternary check 🤔 nor it would be a source for errors. Perhaps adding a small Tsdoc comment would suffice to explain it, no need to really add helper functions or create separate components for this which is relatively small. But for now i will add a comment that will explain it

Reply:
The comment makes it better @ameerul-deriv, but. I still believe this would be much easier to understand, and also eliminates the cognitive complexity.

`/**
 * TSDoc
 */
const RecommendationStatus = ({ recommended }) => {
    if (recommended === null) {
        return null;
    }
    return (<Text as='div' className='flex items-center gap-1' color='less-prominent' size={textSize}>
        {(reviewDetails?.recommended ? (
            <>
                <StandaloneThumbsUpRegularIcon
                    className='mb-[0.3rem]'
                    fill='#4BB4B3'
                    iconSize='sm'
                />
                Recommended
            </>
        ) : (
            <>
                <StandaloneThumbsDownRegularIcon fill='#ec3f3f' iconSize='sm' />
                Not Recommended
            </>
        ))}
</Text>);
 };`